File: Python/Bit Manipulation/Q5_2.py
# Given a real number  between 0 and 1 that is passed in as a double, print the
# binary representation accurately in binary with at most 32 characters, else print
# ERROR


def float_to_binary(number):
    if number > 1.0 or number < 0.0:
        print "Error"
        return

    bin_str = []
    while number > 0.0:
        number *= 2
        if number >= 1.0:
            bin_str.append('1')
            number -= 1
        else:
            bin_str.append('0')

        if len(bin_str) > 32:
            print "Error"

    print "".join(bin_str)


def main():
    float_to_binary(.8125)

if __name__ in "__main__":
    main()

File: Java/Hired Programming Practice/src/BitManipulation/Q5_2.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package BitManipulation;

/**
 *
 * @author marcsantiago
 */
//Given a real number  between 0 and 1 that is passed in as a double, print the
//binary representation accurately in binary with at most 32 characters, else print
//ERROR
public class Q5_2 {
    public static void main(String[] args){
        doubleToBinary(.8125);
    }
    
    public static void doubleToBinary(double number){
        if(number >= 1.0 || number <= 0.0){
            System.out.println("Error");
            return;
        }
           
        StringBuilder sb = new StringBuilder();

        while(number > 0.0){
           number = number * 2;
           if(number >= 1.0){
               number = number - 1.0;
               sb.append(1);
           }
           else{
               sb.append(0);
           }
           
           if(sb.length() > 32){
               System.out.println("Error");
               return;
           }         
        }
        System.out.println(sb.toString());
    }
    
//    public static void intToBinary(int number){
//        StringBuilder sb = new StringBuilder();
//        while(number > 0){
//            int temp = number % 2;
//            number = number / 2;
//            sb.append(temp);
//        }
//        sb.reverse();
//        System.out.println(sb.toString());
//    }
}
